geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -166,6 +120,82 @@ object AccumuloDataStoreFactory {
+    val queryTimeout = queryTimeoutParam.lookupOpt[Int](params).map(i => i * 1000L).orElse {
+      GeomesaSystemProperties.QueryProperties.QUERY_TIMEOUT_MILLIS.option.map(_.toLong)
+    }
+    val collectQueryStats =
+      !connector.isInstanceOf[MockConnector] && collectQueryStatsParam.lookupWithDefault[Boolean](params)
+
+    AccumuloDataStoreConfig(
+      queryTimeout,
+      queryThreadsParam.lookupWithDefault(params),
+      recordThreadsParam.lookupWithDefault(params),
+      writeThreadsParam.lookupWithDefault(params),
+      generateStatsParam.lookupWithDefault[Boolean](params),
+      collectQueryStats,
+      cachingParam.lookupWithDefault(params),
+      looseBBoxParam.lookupWithDefault(params)
+    )
+  }
+
+  def buildAuditProvider(params: JMap[String, Serializable] = EmptyParams) = {
+    security.getAuditProvider(params).getOrElse {
+      val provider = new ParamsAuditProvider
+      provider.configure(params)
+      provider
+    }
+  }
+
+  def buildAuthsProvider(connector: Connector, params: JMap[String, Serializable] = EmptyParams): AuthorizationsProvider = {
+    val forceEmptyOpt: Option[java.lang.Boolean] = forceEmptyAuthsParam.lookupOpt[java.lang.Boolean](params)
+    val forceEmptyAuths = forceEmptyOpt.getOrElse(java.lang.Boolean.FALSE).asInstanceOf[Boolean]
+
+    // convert the connector authorizations into a string array - this is the maximum auths this connector can support
+    val securityOps = connector.securityOperations
+    val masterAuths = securityOps.getUserAuthorizations(connector.whoami)
+    val masterAuthsStrings = masterAuths.map(b => new String(b))
+
+    // get the auth params passed in as a comma-delimited string
+    val configuredAuths = authsParam.lookupOpt[String](params).getOrElse("").split(",").filter(s => !s.isEmpty)
+
+    // verify that the configured auths are valid for the connector we are using (fail-fast)
+    if (!connector.isInstanceOf[MockConnector]) {
+      val invalidAuths = configuredAuths.filterNot(masterAuthsStrings.contains)
+      if (invalidAuths.nonEmpty) {
+        throw new IllegalArgumentException(s"The authorizations '${invalidAuths.mkString(",")}' " +
+          "are not valid for the Accumulo connection being used")
+      }
+    }
+
+    // if the caller provided any non-null string for authorizations, use it;
+    // otherwise, grab all authorizations to which the Accumulo user is entitled
+    if (configuredAuths.length != 0 && forceEmptyAuths) {
+      throw new IllegalArgumentException("Forcing empty auths is checked, but explicit auths are provided")
+    }
+    val auths: List[String] =
+      if (forceEmptyAuths || configuredAuths.length > 0) configuredAuths.toList
+      else masterAuthsStrings.toList
+
+    security.getAuthorizationsProvider(params, auths)
+  }
+
+  def createDataStore(connector: Connector, params: JMap[String, Serializable] = EmptyParams): AccumuloDataStore = {
```

Comment:
since we can pass the connector in through the param map in the other createDataStore method, do we need this at all? Not sure if it's doing something slightly different.